server/server_test.go

```diff
@@ -78,7 +78,9 @@ func TestSecureComm(t *testing.T) {
 	consoleUrl := serverUrl + "/console"
 	apiUrl := serverUrl + "/api"
 
-	server := NewServer(conf)
+	config.Configuration = conf
+
+	server := NewServer()
```

Comment:
you should continue to pass in the config here rather than rely on global var. I think this will make writing unit and i-tests easier (because then we can create servers with different configurations, right?)

I just hate global variables - makes writing tests harder for one reason.

Reply:
you don't like global variables - great.
I don't think it makes sense to propagate a config through 4 or 5 layers of abstraction.
I proposed in my 1st PR a mixed approach, you commented to refactor everything - great.
If tests are executed concurrently we can pass as config but if not I see a contradiction in your comments.

Reply:
So, to summarize the discussion.
It could be good to have a config.Get() and config.Set() pair that will set the configuration global var stored in the config package.